src/Engine/IO/Window.cpp

```diff
@@ -20,30 +19,41 @@ void Window::errorCallback(int error, const char *desc) {
 void Window::keyCallback(GLFWwindow *window, int key, int scancode, int action, int mods) {
     if (key == GLFW_KEY_ESCAPE && action == GLFW_PRESS) {
         glfwSetWindowShouldClose(window, true);
+        return;
     }
 
 #ifdef DEBUG_MODE
-    if (isImGuiEnabled() && (ImGui::IsWindowFocused() || ImGui::IsMouseHoveringAnyWindow())) {
+    if (key == GLFW_KEY_GRAVE_ACCENT && action == GLFW_PRESS) {
+        toggleImGui();
+    }
+    else if (isImGuiEnabled() && (ImGui::IsWindowFocused() || ImGui::IsMouseHoveringAnyWindow())) {
```

Comment:
Should this not be a regular if? Not an else if?

Reply:
the grave key event should be "consumed", and not passed to imgui's callback